src/wp-includes/blocks.php

```diff
@@ -134,17 +134,31 @@ function register_block_script_handle( $metadata, $field_name, $index = 0 ) {
 		$wpinc_path_norm = wp_normalize_path( realpath( ABSPATH . WPINC ) );
 	}
 
-	$theme_path_norm  = wp_normalize_path( get_theme_file_path() );
+	// Cache the $template_path_norm and $stylesheet_path_norm to avoid unnecessary additional calls.
+	static $template_path_norm   = '';
+	static $stylesheet_path_norm = '';
+	if ( ! $template_path_norm || ! $stylesheet_path_norm ) {
+		$template_path_norm   = wp_normalize_path( get_template_directory() );
+		$stylesheet_path_norm = wp_normalize_path( get_stylesheet_directory() );
+	}
 	$script_path_norm = wp_normalize_path( realpath( dirname( $metadata['file'] ) . '/' . $script_path ) );
 
-	$is_core_block  = isset( $metadata['file'] ) && str_starts_with( $metadata['file'], $wpinc_path_norm );
-	$is_theme_block = str_starts_with( $script_path_norm, $theme_path_norm );
+	$is_core_block = isset( $metadata['file'] ) && 0 === strpos( $metadata['file'], $wpinc_path_norm );
+
+	// Determine if the block script was registered in a theme, by checking if the script path starts with either
+	// the parent (template) or child (stylesheet) directory path.
```

Comment:
```suggestion
	/*
	 * Determine if the block script was registered in a theme, by checking if the script path starts with either
	 * the parent (template) or child (stylesheet) directory path.
	 */
```

Use multiline comment